app/services/dropbox/pull_service.rb

```diff
@@ -0,0 +1,100 @@
+module Dropbox
+  class PullService < Operation
+    def execute
+      changes = calculate_changes
+
+      create_or_update(changes[:create] + changes[:update])
+      delete(changes[:delete])
+    end
+
+    private
+
+    def create_or_update(to_create)
+      to_create.each do |entry|
+        path, entry_hsh = entry
+        if entry_hsh['is_dir']
+          create_update_dir(entry_hsh)
+        else
+          download_file(entry_hsh)
+        end
+      end
+    end
+
+    def create_update_dir(entry_hsh)
+      relative_path = remote_to_relative(entry_hsh['path'])
+      dest_path = relative_to_local(relative_path)
+
+      FileUtils.mkdir_p(dest_path) unless File.exist?(dest_path)
+      app_member.dropbox_entries.
+        find_or_initialize_by(path: relative_path).tap do |entry|
+          entry.is_dir = true
+          entry.revision = entry_hsh['rev']
+        end.
+        save
+    end
+
+    def download_file(entry_hsh)
+      tmp = new_tmp_path
+      File.open(tmp, "wb") do |f|
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.